18.09-rc/docker-entrypoint.sh

```diff
@@ -12,9 +12,24 @@ if docker help "$1" > /dev/null 2>&1; then
 	set -- docker "$@"
 fi
 
-# if we have "--link some-docker:docker" and not DOCKER_HOST, let's set DOCKER_HOST automatically
-if [ -z "$DOCKER_HOST" -a "$DOCKER_PORT_2375_TCP" ]; then
-	export DOCKER_HOST='tcp://docker:2375'
+_should_tls() {
+	[ -n "${DOCKER_TLS_CERTDIR:-}" ] \
+	&& [ -s "$DOCKER_TLS_CERTDIR/client/ca.pem" ] \
```

Comment:
If the client is expected to mount the parent directory of `client` (and `server`), the server key is likely to be leaked into the client